app/models/recipe.rb

```diff
@@ -1,3 +1,10 @@
 class Recipe < ApplicationRecord
-   validates :name, :complexity, :genre, presence: true
+  has_many :recipe_ingredients
+  has_many :ingredients, through: :recipe_ingredients
+
+  validates :name, :complexity, :genre, presence: true
+
+  def total_cost
+		ingredients.sum(:cost)
+  end
```

Comment:
Great job abstracting the summing of the cost of the ingredients to a model method and making use of your association.